### PR TITLE
feat(services): migrate configure job setting react query

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-920

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Make it work, make it right, make it fast

File: libs/pages/application/src/lib/feature/page-settings-configure-job-feature/page-settings-configure-job-feature.tsx [hunk 1]
```diff
@@ -1,193 +1,126 @@
...
+  const { data: service } = useService({ serviceId: applicationId, serviceType: 'JOB' })
+  const { mutate: editService, isLoading: isLoadingEditService } = useEditService({ environmentId })
+
+  const schedule = match(service)
+    .with({ job_type: 'CRON' }, (s) => {
+      const { cronjob } = s.schedule
+      return {
+        schedule: cronjob?.scheduled_at,
+        cmd_arguments:
+          cronjob?.arguments && cronjob?.arguments.length > 0 ? JSON.stringify(cronjob?.arguments) : undefined,
```
Comment (ctjhoa) [inline libs/pages/application/src/lib/feature/page-settings-configure-job-feature/page-settings-configure-job-feature.tsx:22]:
```suggestion
        cmd_arguments: JSON.stringify(cronjob?.arguments),
```

Comment (ctjhoa) [reply libs/pages/application/src/lib/feature/page-settings-configure-job-feature/page-settings-configure-job-feature.tsx:22]: 2024-01-05-120017_230x81_scrot

Comment (ctjhoa) [reply libs/pages/application/src/lib/feature/page-settings-configure-job-feature/page-settings-configure-job-feature.tsx:22]: same issue line 31, 37 ,43

File: libs/pages/application/src/lib/feature/page-settings-configure-job-feature/page-settings-configure-job-feature.tsx [hunk 2]
```diff
@@ -1,193 +1,126 @@
...
+    if (!service) return
+
+    try {
+      const schedule = match(service)
+        .with({ job_type: 'CRON' }, () => {
+          return {
+            cronjob: {
+              scheduled_at: data.schedule || '',
+              entrypoint: data.image_entry_point,
+              arguments: data.cmd_arguments ? eval(data.cmd_arguments) : undefined,
```
Comment (ctjhoa) [inline libs/pages/application/src/lib/feature/page-settings-configure-job-feature/page-settings-configure-job-feature.tsx:71]: `eval` is a security issue that should be rewrite
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval#never_use_eval!

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/pages/application/src/lib/feature/page-settings-configure-job-feature/page-settings-configure-job-feature.tsx (hunk 2 above)
Comment (RemiBonnet) [reply libs/pages/application/src/lib/feature/page-settings-configure-job-feature/page-settings-configure-job-feature.tsx:71]: Yes I know this point, I can add comment if you want.

I don't rewrite all behavior but if you want to see the history about why we use `eval`, you can check this thread: https://qovery.slack.com/archives/C02NPSG2HBL/p1664352927296669

File: libs/pages/application/src/lib/feature/page-settings-configure-job-feature/page-settings-configure-job-feature.tsx (hunk 1 above)
Comment (RemiBonnet) [reply libs/pages/application/src/lib/feature/page-settings-configure-job-feature/page-settings-configure-job-feature.tsx:22]: It's why we use eval, you can see the thread bellow

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**